src/sqlfluff/core/linter/linter.py

```diff
@@ -263,22 +263,25 @@ def parse_noqa(comment: str, line_no: int):
 
     @staticmethod
     def remove_templated_errors(
-        linting_errors: List[SQLLintError],
-    ) -> List[SQLLintError]:
+        linting_errors: List[SQLBaseError],
+    ) -> List[SQLBaseError]:
```

Comment:
Parsing "noqa" directives can return `SQLParseError`, so that changes the type of this collection, and we have to update the method to handle the non-`SQLLintError` case.